src/main/java/httpserver/HTTPHandler.java

```diff
@@ -22,232 +22,187 @@
+
+
+    /**
+     * Where the Handler handles the information given from the request and
+     * based off of the paths specified in the Handler. <p>
+     *
+     * This can be overridden for more fine-grained handling. As is, it uses
+     * the data behind the addGET, addPOST, and addDELETE methods for determining
+     * the correct action to take. <p>
+     *
+     * If there is not exact match, the `*` path is used. If you don't have a `*`
+     * catchall route, a 501 (Not implemented) is sent to the client.
+     *
+     * @param request     The incoming HTTPRequest.
+     * @param response    The outgoing HTTPResponse, waiting to be filled by an
+     *                    HTTPHandler.
+     *
+     * @see HTTPHandler#addGET
+     * @see HTTPHandler#addPOST
+     * @see HTTPHandler#addDELETE
+     * @see HTTPResponse#NOT_A_METHOD_ERROR
+     */
+    public void handle(HTTPRequest request, HTTPResponse response) {
+	String httpRequestType = request.getRequestType().toUpperCase();
+	if (!routes.containsKey(httpRequestType)) {
+	    response.message(404, "No " + httpRequestType + " routes exist.");
+	    return;
+	}
+
+	MethodWrapper finalMethod = null;
+	int bestFit = 0;
+	for (MethodWrapper method : routes.get(httpRequestType)) {
+	    if (method.matchesPerfectly(request.getSplitPath())) {
+			finalMethod = method;
+			break;
+	    }
+
+	    int testScore = method.howCorrect(request.getSplitPath());
+	    if (testScore > bestFit) {
+			finalMethod = method;
+			bestFit = testScore;
+			}
+		}
+
+		if (finalMethod == null) {
+			response.message(501, HTTPResponse.NOT_A_METHOD_ERROR);
+			return;
+		}
+
+		finalMethod.invoke(request, response, request.getSplitPath());
+    }
+
+    /**
+     * Attach a method to a GET request at a path. <p>
+     *
+     * Methods are passed in as a String, and must be a member of the current
+     * handler.<p>
+     *
+     * Path's should come in "/path/to/action" form. If the method requires
+     * any parameters that aren't an HTTPResponse, HTTPRequest, or Map,
+     * they should be included in the path, in the order they're
+     * listed in the method header, in "{ClassName}" form. Example:
+     * <code>/hello/{String}/{String}</code> is a good path. <p>
+     *
+     * Methods being passed in must accept an HTTPResponse as their first
+     * parameter. Methods may optionally accept an HTTPRequest and a
+     * Map&lt;String, String&gt; in that order (they may accept a Map but not an
+     * HTTPRequest, but if they accept both the HTTPRequest must come first).
+     *
+     * Parameters following the above must be included in the java.lang library
+     * and have a constructor that takes in a String.
+     * Any other type of parameter will cause an exception to occur. <p>
+     *
+     * Additionally, primitives are not permited, because they're not classes in
+     * the java.lang library. The three most common parameter types are String,
+     * Integer, and Double.
```

Comment:
Almost all of this is inaccurate now. And the future documentation should probably be attached to the `addRoute` method, with the rest of them saying `@see HTTPHandler#addRoute`.

Paths should now come in `/path/to/thing/{variableName}` form, instead of `/path/to/thing/{variableType}`. There's also the option to accept `/path/to/thing/{*}`, which acts as a catchall list (and puts anything in `request.varargs`), which will match like the `{Class...}` matchers did before, but only if it's at the end of the path.

Also, it looks like we never included `{Class...}` documentation...